agent/main.go

```diff
@@ -137,7 +139,7 @@ var (
 )
 
 // TODO - fix logging
-
+// nolint: funlen
```

Comment:
:) IMO this smell should not be suppressed so we are motivated to refactor main as well

Reply:
I agree, should we tackle this in a separate Issue. Needs to put some thoughts on extracting new methods out of main.